e2e/upgrade-testing-rbd.go

```diff
@@ -0,0 +1,148 @@
+			if err != nil {
+				Fail(err.Error())
+			}
+		}
+		createNodeLabel(f, nodeRegionLabel, regionValue)
+		createNodeLabel(f, nodeZoneLabel, zoneValue)
+
+		// fetch current working directory to switch back
+		// when we are done upgrading.
+		cwd, err = os.Getwd()
+		if err != nil {
+			Fail(err.Error())
+		}
+
+		vaultDeploy, err = checkIfVaultDeploymentRequired(upgradeVersion)
+		if err != nil {
+			Fail(err.Error())
+		}
+		if vaultDeploy {
+			deployVault(f.ClientSet, deployTimeout)
+		}
+		err = upgradeAndDeployCSI(upgradeVersion, "rbd")
+		if err != nil {
+			Fail(err.Error())
+		}
+		createConfigMap(rbdDirPath, f.ClientSet, f)
+		createRBDStorageClass(f.ClientSet, f, nil, nil)
+		createRBDSecret(f.ClientSet, f)
+	})
+	AfterEach(func() {
+		if !testRBD || !upgradeTesting {
+			Skip("Skipping RBD Upgrade Testing")
+		}
+		if CurrentGinkgoTestDescription().Failed {
+			// log pods created by helm chart
+			logsCSIPods("app=ceph-csi-rbd", c)
+			// log provisoner
+			logsCSIPods("app=csi-rbdplugin-provisioner", c)
+			// log node plugin
+			logsCSIPods("app=csi-rbdplugin", c)
+		}
+
+		deleteConfigMap(rbdDirPath)
+		deleteResource(rbdExamplePath + "secret.yaml")
+		deleteResource(rbdExamplePath + "storageclass.yaml")
+		deleteVault()
+		if deployRBD {
+			deleteRBDPlugin()
+			if cephCSINamespace != defaultNs {
+				err = deleteNamespace(c, cephCSINamespace)
+				if err != nil {
+					Fail(err.Error())
+				}
+			}
+		}
+		deleteNodeLabel(c, nodeRegionLabel)
+		deleteNodeLabel(c, nodeZoneLabel)
+	})
+
+	Context("Test RBD CSI", func() {
+		It("Test RBD CSI", func() {
+			pvcPath := rbdExamplePath + "pvc.yaml"
+			appPath := rbdExamplePath + "pod.yaml"
+
+			By("checking provisioner deployment is running", func() {
+				err = waitForDeploymentComplete(rbdDeploymentName, cephCSINamespace, f.ClientSet, deployTimeout)
+				if err != nil {
+					Fail(err.Error())
+				}
+			})
+
+			By("checking nodeplugin deamonsets is running", func() {
+				err = waitForDaemonSets(rbdDaemonsetName, cephCSINamespace, f.ClientSet, deployTimeout)
+				if err != nil {
+					Fail(err.Error())
+				}
+			})
+
+			By("upgrade to latest changes and verify app re-mount", func() {
+				_, app := createPVCAndAppBinding(pvcPath, appPath, f, deployTimeout)
+				err = deletePod(app.Name, app.Namespace, f.ClientSet, deployTimeout)
+				if err != nil {
+					Fail(err.Error())
+				}
+				deleteRBDPlugin()
+
+				// support of PSP is available in current master, so we don't need to
+				// skip it now.
+				skipPSP = false
+				err = os.Chdir(cwd)
+				if err != nil {
+					Fail(err.Error())
+				}
+				// deploy vault, if not done during target version deployment.
+				if !vaultDeploy {
+					deployVault(f.ClientSet, deployTimeout)
+				}
+				deployRBDPlugin()
+				// validate if the app gets bound to a pvc created by
+				// an earlier release.
+				err = createApp(f.ClientSet, app, deployTimeout)
+				if err != nil {
+					Fail(err.Error())
+				}
+			})
```

Comment:
lets do a resize and also verify resize of PVC works. and delete PVC at end of this test.

Reply:
Sure. Will add pvc resize check too.

Reply:
Sure. I will add that too.

Reply:
still this is pending